pandas/core/case_when.py

```diff
@@ -0,0 +1,207 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+import numpy as np
+
+from pandas._libs import lib
+
+from pandas.core.dtypes.cast import (
+    construct_1d_arraylike_from_scalar,
+    find_common_type,
+    infer_dtype_from,
+)
+from pandas.core.dtypes.common import is_scalar
+from pandas.core.dtypes.generic import ABCSeries
+
+from pandas.core.construction import array as pd_array
+
+if TYPE_CHECKING:
+    from pandas._typing import Series
+
+
+def case_when(
+    *args: tuple[tuple],
+    default=lib.no_default,
```

Comment:
I think you can tighten up the typing here.  Maybe `args` can be `args: tuple[tuple[Series |  "npt.NDArray[np.bool_]" | Sequence[bool], Sequence | Scalar | Series], ...]`  and `default` can be `Sequence | Scalar | Series`.

Not 100% sure on that.